src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1213,7 +1214,19 @@ static bool addTriggeredSubscriptions_withCache
       LM_T(LmtMongo, ("adding subscription: '%s'", sub.toString().c_str()));
 
       long long throttling       = sub.hasField(CSUB_THROTTLING) ? getField(sub, CSUB_THROTTLING).numberLong() : -1;
-      long long lastNotification = sub.hasField(CSUB_LASTNOTIFICATION) ? getIntField(sub, CSUB_LASTNOTIFICATION) : -1;
+      long long lastNotification = -1;
+
+      if (sub.hasField(CSUB_LASTNOTIFICATION))
```

Comment:
The alternative to BSON() macro (the one which use that `<<` style) is to use BSONObjBuilder and append() method, then use the obj() method to flush to a BSONObj. I guess that BSON() macro (which composes the BSONObj in a shoot) is more efficient, so my recommendation it not stopping using it (in the places it make sense, i.e. for dynamic composition you will still needing BSONObjBuilder) but using it with care.

NTC